org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/ConstraintBasedTypeSystem.xtend

```diff
@@ -152,8 +155,12 @@ class ConstraintBasedTypeSystem implements TypeSystem, TypeProvider {
 	// ************************************************************************
 	// ** Error reporting
 	// ************************************************************************
-	override reportErrors(ConfigurableDslValidator validator) {
-		allVariables.forEach[(it as TypeVariable).reportErrors(validator)]
+	override reportErrors(Resource resource, ConfigurableDslValidator validator) {
+		// Registry will be null if ts is not yet initialized.
+		registry?.allVariables?.forEach [
```

Comment:
This could be more efficient, but it requires some work and I don't think is worth to invest in it right now.